File: Leetcode/easy/Code35.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int start = 0;
        int end = nums.size() - 1;
        int mid = (start + end) / 2;
        while (start <= end) {
            mid = (start + end) / 2;
            if (target == nums[mid]) {
                //nums.insert(nums.begin() + mid, target);
                return mid;
            }
            if (target > nums[mid]) {
                start = mid + 1;
            }
            if (target < nums[mid])
            {
                end = mid - 1;
            }
        }
        if (target > nums[mid]) {
            return mid + 1;
        }
        else {
            return mid;
        }
    }
};

File: Leetcode/easy/src/Code35.java
public class Code35 {
    public int searchInsert(int[] nums, int target) {
        int left = 0;
        int right = nums.length-1;
        int mid = 0;
        while(left<=right){
            mid = left+(right-left)/2;
            if(nums[mid]==target){
                return mid;
            }
            if(nums[mid]<target){
                left = mid+1;
            }
            if(nums[mid]>target){
                right=mid-1;
            }
        }
        return left;
    }
}
